src/Microsoft.Framework.Runtime.Common/Impl/Logger.cs

```diff
@@ -25,17 +34,41 @@ public static void TraceInformation(string message, params object[] args)
 
         public static void TraceWarning(string message, params object[] args)
         {
-            if (IsEnabled)
+            if (IsWarningEnabled)
             {
                 Console.WriteLine("Warning: " + message, args);
             }
         }
 
-        private static bool IsEnabled
+        public static void TraceVerbose(string message, params object[] args)
+        {
+            if (IsVerboseEnabled)
+            {
+                Console.WriteLine("Verbose: " + message, args);
+            }
+        }
+
+        public static bool IsErrorEnabled {  get { return TraceLevel >= InformationLevel; } }
+        public static bool IsWarningEnabled { get { return TraceLevel >= InformationLevel; } }
+        public static bool IsInformationEnabled { get { return TraceLevel >= InformationLevel; } }
+        public static bool IsVerboseEnabled { get { return TraceLevel >= VerboseLevel; } }
+
+        private static int? _traceLevel = null;
+        public static int TraceLevel
         {
             get
             {
-                return Environment.GetEnvironmentVariable(EnvironmentNames.Trace) == "1";
+                if (_traceLevel == null)
+                {
+                    string envVar = Environment.GetEnvironmentVariable(EnvironmentNames.Trace);
+                    int newLevel;
+                    if (string.IsNullOrWhiteSpace(envVar) || !Int32.TryParse(envVar, out newLevel))
+                    {
+                        newLevel = 0;
```

Comment:
Yes, good catch :), I introduced the constants after this code and forgot to put it in here.